docs/lib/views.md

```diff
@@ -48,3 +48,22 @@ class DefaultAPIView(AnsibleBaseView):
     all my good stuff
     ...
 ```
+
+
+## Adding extra related fields
+
+There may be cases where the associative router does not provide a way to add specific related views to your view. As a way to help combat this the `AnsibleBaseView` class has a method defined called `extra_related_fields` which returns an empty dict (`{}`) by default. The entries expected in this dictionary are in the format:
+```
+{
+    "string": "url",
+}
+```
+Where string is the name of the field under the related section in the API and URL is the URL to that object. For example, a return like:
+```
+{
+    'authenticators': reverse('user-authenticators-list', kwargs={'pk': obj.pk}),
+}
+```
+Would cause `authenticators` to be listed under the related field and the related URL will be generated from the user-authenticators-list url and should end up as something like `/api/users/:id/authenticators/`.
+
+This can be particularly useful if a feature in DAB wants to extend a view that is created by a service. In this case you can create a MixIn like `ansible_base.oauth2_provider.views/user_mixin.DABOAuth2UserViewsetMixin` which can set `extra_related_fields` and then have the view in the service extend the MixIn. 
```

Comment:
> This can be particularly useful if a feature in DAB wants to extend a view that is created by a service.

As I said in https://github.com/ansible/django-ansible-base/issues/363, I don't think that's actually a good idea.

> `ansible_base.oauth2_provider.views/user_mixin.DABOAuth2UserViewsetMixin`

`/` -> `.`?

> and then have the view in the service extend the MixIn.

I'd prefer capitalization "Mixin"

And I also don't get it. Wouldn't the app just import a `AuthenticatorViewSet` from DAB and declare that in related views when registering `UserViewSet`? This thing here is really complicated and I don't see what it does that's useful.

Reply:
There's too much ground to cover here. Start with the basics. You have a model in the end-user app, and a related model in DAB. You want to define related endpoints of the end-user's model (User) in the DAB app. You make it sound like you're doing this in DAB, but that's not 100% true even in what you're describing here. You define a mixin _and use that in the end-user's app_. That needs to be spelled out in greater detail. It _is_ defined in the end-user app, what you're describing is just a method for vendoring helper code that the end-user app _can_ use, from DAB, when there is references between those models.